injector/src/main/java/edu/ucr/cs/riple/injector/location/OnField.java

```diff
@@ -58,23 +63,29 @@ protected void fillJsonInformation(JSONObject res) {
   }
 
   @Override
-  protected boolean applyToMember(NodeList<BodyDeclaration<?>> clazz, Change change) {
-    final boolean[] success = {false};
+  protected Modification applyToMember(NodeList<BodyDeclaration<?>> clazz, Change change) {
+    final AtomicReference<Modification> ans = new AtomicReference<>();
```

Comment:
The reason for using `AtomicReference` is to avoid making a single element array in this case. Please see [comment](https://github.com/ucr-riple/NullAwayAnnotator/pull/90#discussion_r1014872217)

Reply:
@lazaroclapp this is just to create a "mutable" local that can be updated inside the nested lambda.  `AtomicReference` is just a common way I've seen to do this; it has nothing to do with concurrency.  If there were a `MutableRef<T>` type we could just use that (but I don't know of one in the standard library, other than something like `WeakReference<T>` which would be even weirder to use).

Reply:
Ok, wasn't familiar with the idiom, but makes sense to me 🙂